### DISCUSSION TITLE
Module-level constants

### DISCUSSION THREAD

**Discussion by jaschdoc** [flix/flix#6874] [Ideas]
# Idea
What do you think of adding support for constants at the top of a namespace as syntactic sugar for a constant function?

```flix
mod MyModule {

    // optionally declare as `pub`
    const one = 1;

    pub def getOne(): Int32 = one;

}
```
desugars to
```flix
mod MyModule {

    // optionally declare as `pub`
    def one(): Int32 = 1;

    pub def getOne(): Int32 = one();

}
```

## Discussion

1. It may be ambiguous whether it refers to a record label or a module constant. Maybe the new parser can handle this?
2. It could be argued that it is just syntactic bloat and you *could easily just add the function yourself*, so the increased complexity of maintaining such a feature for little extra flexibility is not worth it.
3. On the other hand, one could argue that people coming from imperative languages may find it easier to use and it provides them with a safe way of declaring constants.

Additionally, if a constant must be pure (and cannot be recursive) then we could evaluate a constant at compile-time and inline the value (but then we should probably not desugar to a function).

Thoughts?

**Comment by JonathanStarup** [comment]
I think you are right that it is nice for new people. I also think nullary functions have upsides. Nullary functions allow you to decide later that it must now be an actual computation, or it is recursive, or whatever it is. Nullary functions keep the implementation a little more flexible.

I don't think there is a parser confusion since modules are upper case and record operations work on lowercase things.

For the upsides, the compiletime computation could also be done for nullary functions right? see if its "simple enough" to be safe to evaluate or perhaps annote it with `@const` to assert that check.
